[quant][eagermode][qat][test] Add numerical test for qat convert

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47376 [quant][eagermode][qat][test] Add numerical test for qat convert**

Summary:
For leaky_relu

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24734754](https://our.internmc.facebook.com/intern/diff/D24734754)

jerryzh168 committed:
[quant][eagermode][qat][test] Add numerical test for qat convert

Summary:
For sigmoid, hardsimoid, tanh, leaky_relu

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][qat][test] Add numerical test for qat convert"


Summary:
For sigmoid, hardsimoid, tanh, leaky_relu

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24734754](https://our.internmc.facebook.com/intern/diff/D24734754)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][qat][test] Add numerical test for qat convert"


Summary:
For sigmoid, hardsimoid, tanh, leaky_relu

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24734754](https://our.internmc.facebook.com/intern/diff/D24734754)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][qat][test] Add numerical test for qat convert"


Summary:
For sigmoid, hardsimoid, tanh, leaky_relu

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24734754](https://our.internmc.facebook.com/intern/diff/D24734754)

[ghstack-poisoned]

vkuzo commented on torch/quantization/fake_quantize.py:
```diff
@@ -212,8 +212,8 @@ def __init__(self,
         self.quant_max = quant_max
-        self.register_buffer('scale', torch.tensor([scale]))
-        self.register_buffer('zero_point', torch.tensor([zero_point]))
+        self.register_buffer('scale', torch.tensor([scale], dtype=torch.float))
+        self.register_buffer('zero_point', torch.tensor([zero_point], dtype=torch.float))
```
qq, what's the context for this one?

jerryzh168 replied:
added when debugging, just to make it more explicit, is this needed?

vkuzo replied:
makes sense, I was just wondering, lg

jerryzh168 replied:
update: removed for now, looks like we didn't do that in fake_quantize.py right now, maybe we can add this back when we have better numerical test coverage on the qat conversion later.

jerryzh168 committed:
Update on "[quant][eagermode][qat][test] Add numerical test for qat convert"


Summary:
For leaky_relu

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24734754](https://our.internmc.facebook.com/intern/diff/D24734754)

[ghstack-poisoned]
